code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceConsumeMapDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package io.cattle.platform.servicediscovery.dao.impl;
+
+import static io.cattle.platform.core.model.tables.ServiceConsumeMapTable.SERVICE_CONSUME_MAP;
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.model.ServiceConsumeMap;
+import io.cattle.platform.core.model.tables.records.ServiceConsumeMapRecord;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.servicediscovery.api.dao.ServiceConsumeMapDao;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class ServiceConsumeMapDaoImpl extends AbstractJooqDao implements ServiceConsumeMapDao {
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Override
+    public ServiceConsumeMap findMapToRemove(long serviceId, long consumedServiceId) {
+        ServiceConsumeMap map = objectManager.findOne(ServiceConsumeMap.class,
```

Comment:
You'll have to add the removed filter to both of these calls: findMapToRemove() and findNonRemovedMap()